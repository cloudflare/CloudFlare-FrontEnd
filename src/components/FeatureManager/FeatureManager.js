import React, { Component } from 'react';

export default class FeatureManager extends Component {
  render() {
    return (
      <div>
        {this.props.isEnabled && this.props.children}
        {!this.props.isEnabled && this.props.error && this.props.error}
      </div>
    );
  }
}

FeatureManager.propTypes = {
  isEnabled: React.PropTypes.bool.isRequired,
  error: React.PropTypes.string
};
