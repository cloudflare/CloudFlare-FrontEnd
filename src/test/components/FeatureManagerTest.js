import React from 'react';
import FeatureManager from '../../components/FeatureManager/FeatureManager';
import renderer from 'react-test-renderer';

it('renders enabled correctly', () => {
  const component = renderer
    .create(
      <FeatureManager isEnabled={true} error="error">children</FeatureManager>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});

it('renders disabled correctly', () => {
  const component = renderer
    .create(
      <FeatureManager isEnabled={false} error="error">children</FeatureManager>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
