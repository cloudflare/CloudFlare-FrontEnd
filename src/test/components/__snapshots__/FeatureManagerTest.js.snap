// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders disabled correctly 1`] = `
<div>
  error
</div>
`;

exports[`renders enabled correctly 1`] = `
<div>
  children
</div>
`;
